CONTRIBUTING.md

```diff
@@ -55,13 +55,19 @@ The `changelog.py` script is used to generate the final changelog (it's not perf
 - Use _rebase workflow_ for all PRs.
 - When addressing review comments, if possible, update original commits with fixes belonging to them. Then force-push.
 
+## Naming Conventions
+- Type names are in `PascalCase`.
+- Global constants and enum items are in `SCREAMING_CASE`.
+- Variables (local, global, struct members) and functions are in `snake_case`.
+- File names are in `snake_case.c`.
+
 ## CI
 CI runs on every push to main and needs to pass on every PR. It:
 - Runs a build of the package.
 - Checks formatting of C sources using `clang-format`.
 
 ## Code Formatting
-C code is automatically formatted using `clang-format` version 18 (you need to provide this dependency yourself).
+C code is automatically formatted using `clang-format` version 18. You will need to provide this dependency yourself. See: [LLVM builds page](https://llvm.org/builds/) and [LLVM clang docs for integrations](https://releases.llvm.org/18.1.6/tools/clang/docs/ClangFormat.html)
```

Comment:
I'm not sure how helpful those links are. I would expect most people to install `clang-format` primarily from their distro and I don't want to encourage downloading precompiled binaries, as IMO they should be the last resort thing, not first (I do wish there was a standard and automated way to specify and pull those dependencies, but there is not...).

And then for the integrations link, one thing is it's tied to a version so it adds to maintenance burden for when the version is updated. But more importantly I find it of little relevance to bootstrapping the build environment for the package itself. It contains various integration guidelines which are not really relevant to the package itself, you could add links to dozens of other useful setup articles if you wanted to go that way, but I don't think it belongs here.

Reply:
Removed any references to a version, and instead linked to the base releases page (Which itself includes links to binaries, builds, and docs)

My thinking here is this: If someone is the type of person who wants to build their own packages, they will seek out the source for doing so. However, if someone is more interested in diving into getting started with helping in development here, lowering the bar to entry to doing so makes that easy and inviting! That seems like a good thing to move toward. If the concern is with encouraging download of precompiled packages, perhaps I can add a note about that as a "best practice", but I think many coming to this page would be aware already.

Reply:
Well, I don't like just linking to the overall download page, because that makes it look like downloading the binary is the recommended/expected way to do it, while to me it's the very last resort. I know some people don't care too much, but it's technically a bad practice.

Also, these are contributing guidelines, it's expected you know your way around if you want to contribute code. And if not, you're willing to research it and figure it out.

Anyway, if you really want to add to this section (while, at the same time, not making it too long and bloat the contribution guidelines), please:
- List the packages to install on the common distros (Ubuntu and Fedora I guess). This is a bit complicated because sometimes the package is called `clang-tools` or `clang-tools-extra` (so this can be helpful information, but requires verifying the package names). I think the latter is the name for Fedora and I'm not entirely sure what's the situation on Ubuntu atm., I found booth the clang-tools and clang-format packages via a quick search. They also may or may not have a -18 in their name.
- Mention clang-format can be pulled as a dev environment dependency via other means.
- If nothing else is available, download the binary.

Reply:
> Also, these are contributing guidelines, it's expected you know your way around if you want to contribute code. And if not, you're willing to research it and figure it out.

Just trying to make it more approachable. I can see this isn't actually helpful, though. I think I'll just close this for the time being.

Reply:
Some of the changes in this PR were helpful improvements...